### ISSUE TITLE
Edit File Tags: Duplicate Custom Tags

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#2991]
After working on the UI improvements for Edit File Tags in #2391, it was again suggested that we should review the possibility of consolidating the code for the two version of the Edit File Tags popup, on the Dataset pg and the Edit File pg.

In addition to consolidating the two popups, there are a few remaining bugs that also need to be addressed in these two popups:

Dataset Pg > Edit Tags Popup
- ~Remove a File Tag by unchecking it in the dropdown, then add a new Custom File Tag, click "Apply" button and it is added to the Selected File Tags, but the File Tag that was previously removed is added back.~

Edit Files Pg > Edit Tags Popup
- ~Delete Tags checkbox was not included in this popup due to the complicated "Edit Files-Save Changes" workflow and development questions that were still unanswered.~
- ~Add Custom File Tag in popup for one file, do not see Custom File Tag in popup for second file unless you “Save Changes” on Edit File pg, and return to the Edit Files version of the popup a second time.~

UPDATED COMMENT ON 6/18/19 by @mheppler, see additional info below

**Comment by mheppler** [comment]
Attempted to replicate the "minor" bugs reported above, but they seemed to have been resolved. Could be fix by a commit from @sekmiller on Mar 18, 2016 6f02f7e4d890a78dd9d82c2c2c17fb5af9cf809f that was part of the PR for said #2391 linked in the comment above.

One new bug for the Edit Tags popup that I discovered in demo (v. 4.14 build 1326-57a392d), on both the dataset and edit files pgs, is that when you create a custom file tag for one file, and then add it to a second in the same edit file metadata workflow, it results in two custom tags being generated and displayed in the dropdown going forward.

![Screen Shot 2019-06-18 at 11 00 44 AM](https://user-images.githubusercontent.com/687227/59695639-8ac9c600-91b8-11e9-813e-35298565b4d5.png)

Also, to simplify this issue, I have removed the "consolidate two popups" suggestion from the title, as that is a far more involved code refactoring.
